src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -154,7 +154,7 @@ public function decode(string $data, string $format, array $context = []): mixed
                     $headerCount = array_fill(0, $nbCols, 1);
                 } else {
                     foreach ($cols as $col) {
-                        $header = explode($keySeparator, $col ?? '');
+                        $header = false === $keySeparator ? [$col] : explode($keySeparator, $col ?? '');
```

Comment:
Don't forget to update the `CsvEncoderContextBuilder` so that it can handle the `false` value.